File: Card.java
/****************************************************************************
 * Created by: Shuvaethy Neill
 * Created on: May 2019
 * Created for: ICS4U
 * This program is the game of 21
 ****************************************************************************/

public class Card
{
	public int value;
	public String name;
	public Boolean isAvailableToDeal;
	
	public Card(String n, int v){
		name = n;
		value = v;
		isAvailableToDeal = true;
	}

	public Boolean IsCardAvailableToDeal() {
		return isAvailableToDeal;
	}

	public void SetCardHasBeenDealt() {
		isAvailableToDeal = false;
	}

	public void SetCardIsAvailableToDeal() {
		isAvailableToDeal = true;
	}

	public String toString() {
		return "Card: " + name;
	}
}

File: Card.py
class Card:

  def __init__(self, n, v):
    self._name = n
    self._value = v
    self._isAvailableToDeal = True

  def isCardAvailableToDeal(self):
    return self._isAvailableToDeal
  
  def setCardHasBeenDealt(self):
    self._isAvailableToDeal = False

  def setCardIsAvailableToDeal(self):
    self._isAvailableToDeal = True

  def toString(self):
    return self._name
  
  def isAce(self):
    return self._name[0] == 'A'

  def getValue(self):
    return self._value
